### PR TITLE
Add resource management alerts

### PR DESCRIPTION
**Pull request by glrf**
This PR adds capacity planing alerts. It adds alerts for pod count as well as CPU/memory requests and actual utilization for worker nodes.

The actual alerts might need some tweaking, but should be easily done through the parameters.

This took a lot longer than anticipated and I probably went a bit overboard, but I did not want to throw in some half baked alerts.

## Checklist

- [x] PR contains a single logical change (to build a better changelog).
- [x] Update the documentation.
- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog.

### CODE REVIEW

File: docs/modules/ROOT/pages/explanations/resource_management.adoc
Comment (corvus-ch) [inline docs/modules/ROOT/pages/explanations/resource_management.adoc:275]:
```suggestion
It's hard to say what level of overprovisioning is OK and what's not, so observing the actual resource usage is more effective.
```

**Review by corvus-ch — APPROVED** [review]
Looking good to me. Please await a second approval before merging.

File: component/capacity.libsonnet
Comment (simu) [inline component/capacity.libsonnet:49]: You'll want to introduce `memoryAllocatable` and `cpuAllocatable` for these alerts, as discussed in my other comment.

File: docs/modules/ROOT/pages/explanations/resource_management.adoc
Comment (simu) [inline docs/modules/ROOT/pages/explanations/resource_management.adoc:74]:
```suggestion
They alert if there isn't enough CPU or memory capacity if the largest worker node disappears or if we predict that the cluster will exceed this threshold in three days.
```

File: docs/modules/ROOT/pages/runbooks/cpucapacity.adoc
Comment (simu) [inline docs/modules/ROOT/pages/runbooks/cpucapacity.adoc:26]:
```suggestion
If the load reported by `top nodes` doesn't appear to be particularly high, there might be a bug in the alert rule.
```

File: docs/modules/ROOT/pages/runbooks/cpucapacity.adoc
Comment (simu) [inline docs/modules/ROOT/pages/runbooks/cpucapacity.adoc:56]:
```suggestion
This alert indicates that we expect the total CPU utilization over all worker nodes to become high in the next days.
```

File: docs/modules/ROOT/pages/runbooks/cpucapacity.adoc
Comment (simu) [inline docs/modules/ROOT/pages/runbooks/cpucapacity.adoc:59]:
```suggestion
If you receive this alert, the cluster might soon experience degraded workload performance in case of a worker node failure.
```

File: docs/modules/ROOT/pages/runbooks/memorycapacity.adoc
Comment (simu) [inline docs/modules/ROOT/pages/runbooks/memorycapacity.adoc:10]:
```suggestion
This alert indicates that the total memory usage over all worker nodes is high.
```

File: docs/modules/ROOT/pages/runbooks/memorycapacity.adoc
Comment (simu) [inline docs/modules/ROOT/pages/runbooks/memorycapacity.adoc:13]:
```suggestion
If you receive this alert, a worker node failure in the cluster will likely cause customer workloads to be OOM killed.
```

File: docs/modules/ROOT/pages/runbooks/memorycapacity.adoc
Comment (simu) [inline docs/modules/ROOT/pages/runbooks/memorycapacity.adoc:27]:
```suggestion
If that's the case, please disable this alert and open an issue for this component.
```

File: docs/modules/ROOT/pages/runbooks/memorycapacity.adoc
Comment (simu) [inline docs/modules/ROOT/pages/runbooks/memorycapacity.adoc:33]:
```suggestion
If this alert isn't actionable, noisy, or was raised too late you might want to tune it.
```

File: docs/modules/ROOT/pages/runbooks/podcapacity.adoc
Comment (simu) [inline docs/modules/ROOT/pages/runbooks/podcapacity.adoc:6]:
```suggestion
We provide two rules to notice if a cluster is hitting the limit of schedulable pods.
```

File: docs/modules/ROOT/pages/runbooks/podcapacity.adoc
Comment (simu) [inline docs/modules/ROOT/pages/runbooks/podcapacity.adoc:15]:
```suggestion
If you receive this alert, it's likely that the cluster won't be able to keep all workloads running in case of a worker node failure.
```

File: docs/modules/ROOT/pages/runbooks/podcapacity.adoc
Comment (simu) [inline docs/modules/ROOT/pages/runbooks/podcapacity.adoc:16]:
```suggestion
After verifying that the amount of running pods in the cluster is close to the cluster's pod capacity, you should add worker nodes as soon as possible.
```

File: docs/modules/ROOT/pages/runbooks/podcapacity.adoc
Comment (simu) [inline docs/modules/ROOT/pages/runbooks/podcapacity.adoc:34]:
```suggestion
If there's a discrepancy between the alert, and the actual number of running pods reported by Kubernetes, there might be a bug in the alert rule.
```

File: docs/modules/ROOT/pages/runbooks/resourcerequests.adoc
Comment (simu) [inline docs/modules/ROOT/pages/runbooks/resourcerequests.adoc:26]:
```suggestion
If the allocated resources reported by Kubernetes don't indicate high memory reservation on the nodes, there might be a bug in the alert rule.
```

File: docs/modules/ROOT/pages/runbooks/resourcerequests.adoc
Comment (simu) [inline docs/modules/ROOT/pages/runbooks/resourcerequests.adoc:33]:
```suggestion
If this alert isn't actionable, noisy, or was raised too late you might want to tune it.
```

File: docs/modules/ROOT/pages/runbooks/resourcerequests.adoc
Comment (simu) [inline docs/modules/ROOT/pages/runbooks/resourcerequests.adoc:57]:
```suggestion
By default, it will fire if the predicted amount of unrequested memory in three days is less than the capacity of the largest worker node.
```

File: docs/modules/ROOT/pages/runbooks/resourcerequests.adoc
Comment (simu) [inline docs/modules/ROOT/pages/runbooks/resourcerequests.adoc:134]:
```suggestion
If this alert isn't actionable, noisy, or was raised too late you might want to tune it.
```

File: docs/modules/ROOT/pages/runbooks/resourcerequests.adoc
Comment (simu) [inline docs/modules/ROOT/pages/runbooks/resourcerequests.adoc:157]:
```suggestion
This alert indicates that we expect the total sum of CPU requests of user workloads to soon be close to the capacity of the cluster.
```

File: docs/modules/ROOT/pages/runbooks/resourcerequests.adoc
Comment (simu) [inline docs/modules/ROOT/pages/runbooks/resourcerequests.adoc:160]:
```suggestion
If you receive this alert, the cluster may soon not be able to reschedule all workloads in case of a worker node failure.
```

File: docs/modules/ROOT/pages/runbooks/resourcerequests.adoc
Comment (simu) [inline docs/modules/ROOT/pages/runbooks/resourcerequests.adoc:161]:
```suggestion
After verifying that the cluster's CPU reservation is growing, you should add more CPU capacity in the next days.
```

File: docs/modules/ROOT/pages/explanations/resource_management.adoc
Comment (simu) [inline docs/modules/ROOT/pages/explanations/resource_management.adoc:259]:
```suggestion
By default, we use the one day moving average over one day for queries which predict three days into the future.
```

File: docs/modules/ROOT/pages/explanations/resource_management.adoc
Comment (simu) [inline docs/modules/ROOT/pages/explanations/resource_management.adoc:262]:
```suggestion
For example, if an environment is updated using a blue-green deployment, without the moving average the alert will see this sudden increase in resource usage, extrapolate the increase over three days and will fire, generating a false-positive alert.
```

File: docs/modules/ROOT/pages/runbooks/cpucapacity.adoc
Comment (simu) [inline docs/modules/ROOT/pages/runbooks/cpucapacity.adoc:57]:
```suggestion
By default, this alert will fire if we expect the number of idle cores to less than the core count of the largest worker node in three days.
```

File: docs/modules/ROOT/pages/runbooks/memorycapacity.adoc
Comment (simu) [inline docs/modules/ROOT/pages/runbooks/memorycapacity.adoc:11]:
```suggestion
By default, it will fire if the amount unused memory is less than the memory capacity of the largest worker node.
```

File: docs/modules/ROOT/pages/runbooks/memorycapacity.adoc
Comment (simu) [inline docs/modules/ROOT/pages/runbooks/memorycapacity.adoc:57]:
```suggestion
By default, this alert will fire if we expect the amount of unused memory to be less than the memory capacity of the largest worker node in three days.
```

File: docs/modules/ROOT/pages/runbooks/podcapacity.adoc
Comment (simu) [inline docs/modules/ROOT/pages/runbooks/podcapacity.adoc:13]:
```suggestion
By default, this threshold is the pod limit per node.
```

File: docs/modules/ROOT/pages/runbooks/resourcerequests.adoc
Comment (simu) [inline docs/modules/ROOT/pages/runbooks/resourcerequests.adoc:11]:
```suggestion
By default, it will fire if the amount of unrequested memory is less than the capacity of the largest worker node.
```

**Review by simu — CHANGES_REQUESTED** [review]
Implementation looks good. As noted inline, we should use `kube_node_status_allocatable` for the alerts looking at CPU/memory requests.

Other than that, just a bunch of documentation comments.

File: class/defaults.yml
Comment (glrf) [reply class/defaults.yml:144]: That's a very good input. I did not know that. Thanks!

**Review by simu — APPROVED** [review]

**Pull request merged**
